app/models/purchase.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+const { databaseError } = require('../errors');
+
+module.exports = (sequelize, DataTypes) => {
+  const Purchase = sequelize.define(
+    'purchase',
+    {
+      id: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true
+      },
+      userId: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        field: 'user_id'
+      },
+      albumId: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        field: 'album_id'
+      }
+    },
+    { underscored: true }
+  );
+
+  Purchase.buyAlbumWithUserId = (userId, albumId) =>
+    Purchase.findOrCreate({
+      where: { userId, albumId }
+    }).catch(error => databaseError(error.message));
+
+  Purchase.findPurchase = (userId, albumId) =>
+    Purchase.findOne({ where: { userId, albumId } }).catch(error => databaseError(error.message));
+  /* albums.associate = function(models) {
+    // associations can be defined here
+  };*/
```

Comment:
We usually leave this commented after the `define` but before the model functions, to keep order between model configuration and model functions. Also, the comment is a bit outdated, the model is named pruchases now 😝

Reply:
Should it be purchases or purchase?
That comment got updated in the user model too. So I don't know which one should I use.
`  /* User.associate = function(models) {
    // associations can be defined here
  };*/`

Reply:
Whoops, I had an extra s. Should be purchase!